(doc) Use parameter and argument appropriately in Options (#1934)
ocaml/ocaml.org: data/tutorials/language/3ds_00_options.md

```

The types of `f` and `g` are reversed. Functions `Option.fold` and `option_unfold_try` also work in a reversed manner

> * `Option.fold` has as an option argument

Reviewer: ```suggestion
* `Option.fold` has as an option parameter
```

> * `Option.fold` takes an option argument to build something

Reviewer: ```suggestion
* `Option.fold` has as an option argument
```

> * `option_unfold_try` returns an option result from some argument

Reviewer: ```suggestion
* `option_unfold_try` returns an option result

The standard library does not contain any option unfolding function.

### Bind an Option

> The `bind` function of type `'a option -> ('a -> 'b option) -> 'b option` works a bit like `map`. But whilst `map` expects a function-as-argument `f` that returns an unwrapped value of type `b`, `bind` expects an `f` that returns a value already wrapped in an option `'b option`.

Reviewer: ```suggestion
The `bind` function of type `'a option -> ('a -> 'b option) -> 'b option` works a bit like `map`. But whilst `map` expects a function parameter `f` that returns an unwrapped value of type `b`, `bind` expects an `f` that returns a value already wrapped in an option `'b option`.
```

val bind : 'a option -> ('a -> 'b option) -> 'b option = <fun>
```

> Observe that the types are the same, except for the codomain of the function-as-argument.

Reviewer: ```suggestion
Observe that the types are the same, except for the codomain of the function parameter.
```
